refactor(home): migrate view to react [EE-1810]

chiptus opened the pull request:
[EE-1810]

[EE-1810]: https://portainer.atlassian.net/browse/EE-1810?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

chiptus committed:
refactor(home): migrate view to react [EE-1810]

fixes [EE-1810]

refactor(home): wip use react view

feat(home): show message if no endpoints

refactor(home): show endpoint list

refactor(home): don't use home to manage link

refactor(home): move state

refactor(home): check if edge using util

refactor(home): move inf panels

chore(home): tests

refactor(home): load groups and tags in env-item

refactor(settings): revert publicSettings change

refactor(home): move confirm snapshot method

chiptus committed:
feat(app): ui-state context provider

chiptus committed:
refactor(app): create InformationPanel

chiptus committed:
feat(endpoints): fetch number of total endpoints

cheloRydel commented on app/portainer/environments/environment.service/index.ts:
```diff
@@ -0,0 +1,295 @@
+  } catch (e) {
+    const axiosError = e as AxiosError;
+
+    if (!axiosError.isAxiosError) {
```
I'm not a big fan of this particular check being repeated multiple times, would it make sense to implement a helper function, maybe in the `axios.ts` file something like `parseAxiosError` and here do something shorter like
`catch(e) {
	throw parseAxiosError(e);
}`

cheloRydel commented on app/portainer/services/api/endpoint.service.ts:
```diff
@@ -0,0 +1,50 @@
+import {
```
shouldn't this file be moved to `app/portainer/environments` ? or is it here for backwards compatibility for now?

chiptus replied:
backward compatibility

chiptus committed:
refactor(http): parse axios errors (#6325)

chiptus committed:
refactor(home): use endpoint-list as react component [EE-1814] (#6060)

* refactor(home): use endpoint-list as react component

fix(home): add missing features and refactors

- kubebutton
- group name
- poll when endpoint is off
- state management

refactor(endpoints): use stat component

fix(endpoints): add space between items

refactor(endpoints): move stats to components

refactor(endpoints): fetch time

refactor(home): move logic

refactor(home): move fe render logic

refactor(settings): use vanilla js for publicSettings

refactor(kube): remove angular from kube config service

feat(home): add kubeconfig button

feat(home): send analytics when opening kubeconfig modal

fix(home): memoize footer

refactor(home): use react-query for loading

fix(home): show correct control for kubeconfig modal

refactor(home): use debounce

refactor(home): use new components

refactor(home): replace endpoints with environments

refactor(home): move endpoint-list component to home

fix(home): show group name

refactor(home): use switch for environment icon

fix(kubeconfig): fix default case

refactor(axios): use parse axios error

refactor(home): use link components for navigate

fix(home): align azure icon

refactor(home): refactor stats

refactor(home): export envstatusbadge

refactor(home): remove unused bindings

* chore(home): write tests for edge indicator

* chore(home): basic stories for environment item

* style(settings): reformat

* fix(environments): add publicurl

* refactor(home): use table components

* refactor(datatables): merge useSearchBarState

* refactor(home): fetch group in env item

* chore(tests): basic tests

* chore(home): test when no envs

chiptus committed:
refactor(app): merge hooks

chiptus committed:
refactor(tags): use axios for tagService

chiptus committed:
refactor(env-groups): use axios for getGroups

chiptus committed:
refactor(home): create MotdPanel

xAt0mZ reviewed (approved):
LGTM

chiptus committed:
fix(home): show tags

chiptus committed:
fix(environments): handle missing snapshots

chiptus committed:
fix(kube/volumes): fetch pesistent volume claims

chiptus committed:
refactor(endpoints): set current endpoint

chiptus committed:
refactor(kube): remove use of endpointProvider

chiptus committed:
chore(home): add data-cy for tests

chiptus committed:
chore(tests): mock axios-progress-bar

itsconquest commented:
/azp run

itsconquest commented:
/azp run

chiptus commented:
/azp run

chiptus committed:
refactor(home): move use env list to env module

chiptus commented:
/azp run

itsconquest commented:
/azp run

chiptus committed:
feat(app): sync home view changes with ee

chiptus committed:
fix(home): sort page header

chiptus committed:
fix(app): fix tests

chiptus committed:
chore(github): use yarn cache

chiptus committed:
refactor(environments): load list of groups

chiptus committed:
chore(babel): remove auto 18n keys extraction

chiptus committed:
chore(environments): fix tests

chiptus committed:
refactor(k8s/application): use current endpoint

chiptus committed:
fix(app/header): add margin to header

chiptus committed:
refactor(app): remove unused types

chiptus committed:
refactor(app): use rq onError handler

chiptus committed:
refactor(home): wrap element with button

chiptus commented:
/azp run

huib-portainer commented:
Closes https://github.com/portainer/portainer/issues/6031
